pre_commit/languages/lua.py

```diff
@@ -0,0 +1,137 @@
+import contextlib
+import os
+import re
+from typing import Generator
+from typing import Sequence
+from typing import Tuple
+
+from pre_commit.envcontext import envcontext
+from pre_commit.envcontext import PatchesT
+from pre_commit.envcontext import Var
+from pre_commit.hook import Hook
+from pre_commit.languages import helpers
+from pre_commit.prefix import Prefix
+from pre_commit.util import clean_path_on_failure
+from pre_commit.util import cmd_output
+
+ENVIRONMENT_DIR = 'lua_env'
+get_default_version = helpers.basic_get_default_version
+healthy = helpers.basic_healthy
+
+
+def _find_lua() -> str:  # pragma: win32 no cover
+    """Find a lua executable.
+
+    Lua doesn't always have a plain `lua` executable.
+    Some OS vendors will ship the binary as `lua#.#` (e.g., lua5.3)
+    so discovery is needed to find a valid executable.
+    """
+    paths = os.environ.get('PATH', '').split(os.pathsep)
+    possible_luas = []
+    for path in paths:
+        try:
+            candidate_executables = os.listdir(path)
+        except FileNotFoundError:
+            # Invalid path on PATH.
+            continue
```

Comment:
this can be `with contextlib.suppress(FileNotFoundError):` -- though I suspect this should just suppress all `OSError`s since there's possibility for `PermissionError` as well

Reply:
This try/except is in the for loop so I'm not sure how the code could continue if the exception was suppressed. I think if I used `suppress` and the check failed, then the code could hit an `UnboundLocalError`.

Good call on the `PermissionError`. I'll switch to catching `OSError`.

Reply:
oh I see now, I think I missed the indentation